test/test_gpu2.cpp

```diff
@@ -9201,84 +9201,83 @@ TEST_F(NVFuserTest, FusionTestWarpSoftMax_CUDA) {
-//     if (auto alloc = dynamic_cast<kir::Allocate*>(kir_node)) {
-//       auto size = alloc->size();
-//       if (!(alloc->buffer()->name() == 1 || alloc->buffer()->name() == 2)) {
-//         // There should be no allocation other than those for tv1 and tv2 and
-//         // hoisted indices
-//         TORCH_CHECK(
-//             alloc->buffer()->isIntegralScalar() ||
-//             alloc->buffer()->isABool(), "Invalid allocation detected");
-//       }
-//       TORCH_CHECK(size->isConst(), "Allocation not constant");
-//       auto size_int = size->value();
-//       if (alloc->buffer()->name() == 1) {
-//         TORCH_CHECK(
-//             size_int == split_factor,
-//             "Invalid allocation size: ",
-//             size->value());
-//         tv1_validated = true;
-//       } else {
-//         TORCH_CHECK(size_int == 1, "Invalid allocation size: ",
-//         size->value()); tv2_validated = true;
-//       }
-//     }
-//   }
-//
-//   TORCH_CHECK(tv1_validated, "Failed to validate tv1 allocation");
-//   TORCH_CHECK(tv2_validated, "Failed to validate tv2 allocation");
-//
-//   auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
-//   at::Tensor t0 = at::randn({99, 101}, options);
-//   std::vector<c10::IValue> aten_inputs = {t0};
-//
-//   FusionExecutor fe;
-//   fe.compileFusion(&fusion, aten_inputs);
-//   auto outputs = fe.runFusion(aten_inputs);
-//
-//   auto ref = (t0 + 1).sum({1}) + 1;
-//
-//   testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
-// }
+TEST_F(NVFuserTest, FusionIssue1133_CUDA) {
+  if (!deviceMajorMinorCheck(7)) {
+    GTEST_SKIP() << "skipping tests on pre-Volta GPUs";
+    return;
+  }
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, IrBuilder::create<Val>(1.0));
+  auto tv2 = sum(tv1, {1});
+  auto tv3 = add(tv2, IrBuilder::create<Val>(1.0));
+
+  fusion.addOutput(tv3);
+
+  tv0->computeAt(tv3, 1);
+
+  const int split_factor = 32;
+
+  tv2->split(-1, split_factor);
+  tv1->computeAt(tv2, -2);
+
+  tv1->axis(-1)->parallelize(ParallelType::TIDx);
+  tv2->axis(-1)->parallelize(ParallelType::TIDx);
+
+  tv3->axis(0)->parallelize(ParallelType::Unswitch);
+
+  tv1->setMemoryType(MemoryType::Shared);
+  tv2->setMemoryType(MemoryType::Shared);
+
+  // Both tv1 and tv2 should be allocated at the top-level scope
+  GpuLower gpulw(&fusion);
+  bool tv1_validated = false;
+  bool tv2_validated = false;
+  for (const auto& kir_node : gpulw.kernel()->topLevelExprs()) {
+    if (auto alloc = dynamic_cast<kir::Allocate*>(kir_node)) {
+      auto size = alloc->size();
+      if (!(alloc->buffer()->name() == 1 || alloc->buffer()->name() == 2)) {
+        // There should be no allocation other than those for tv1 and tv2 and
+        // hoisted indices
+        TORCH_CHECK(
+            !alloc->buffer()->isA<TensorView>(), "Invalid allocation detected");
```

Comment:
I wonder why this needs to change.

Reply:
Because in the past, indices and predicates only contain int/bool subexpressions, so index hoisting only creates int or bool allocations. But now, it can also create arrays and structs.

Reply:
No, I ran the test when it was first reported to fail on our CI. Maybe some of the recent changes broke the assertion?

Reply:
It is the change in this PR breaks the assertion. Because we started generating structured stride access, so now index hoisting start to hoist structs and arrays.